packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -422,23 +446,55 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
     function withdrawProposalRewards(address perpetual) external nonReentrant() publishAndWithdrawProposal(perpetual) {}
 
     /**
-     * @notice Set the reward rate (per second) for a specific `perpetual` contract.
-     * @dev Callable only by the Perpetual contract.
+     * @notice Set new parameters for a specific `perpetual` contract. New params go into effect
+     * after a liveness period passes.
+     * @dev Callable only by owner. Calling this while there is already a pending RecordParams struct
+     * will overwrite it.
      */
-    function setRewardRate(address perpetual, FixedPoint.Unsigned memory rewardRate)
+    function setRecordParams(address perpetual, RecordParams memory newRecordParams)
+        external
+        onlyOwner()
```

Comment:
yep. the deployer of the perp would be the factory, what we could do is use MultiRole to be more flexible. Right now the deployer of the perp doesn't have any permissions anywhere else in the contract so I didn't think that was the natural owner

Reply:
so who is responsible for setting this & how does multi-access work on this? Is this a DVM parameter or some external DVM logic? If it was a MultiRole who would be included?

Reply:
If we used MultiRole, then there'd still be some owner who had permission to add new roles, this would presumably be the FRS' deployer. If it was multirole, I imagine, each Perp's deployer would be added for that specific perpetual, but it gets tricky. 

It might be easier if this were all configurable on the Perpetual contract's side.